### ISSUE TITLE
useChain breaks useTrail behaviour

### ISSUE THREAD

**Issue by mrgnou** [pmndrs/react-spring#1512]
## 🐛 Bug Report

When using useChain with useTrail references, trail animations are played all at once. 
This issue can possibly be related with #1492. 

## Expected behavior

useTrail should keep playing staggered animations while used with useChain.

## Link to repro

[Codesandbox](https://codesandbox.io/s/elegant-noyce-mc4rd?file=/src/App.tsx:0-1427)

## Environment

- `react-spring` v9.1.2
- `react` v17.0.2

**Comment by Emiliano-Bucci** [comment]
Is there any way to workaraound it for the moment?

**Comment by thexpand** [comment]
In the meantime, I figured out a workaround for this issue.

If you want to chain them one after another (without overlapping), you can make all but the first one paused and then resume the animation in the `onRest` event callback like this:
```js
const [trail, trailApi] = useTrail(trailLength, () => ({
  // Your trail props go here...
  pause: true,
}));

const [springStyles, springApi] = useSpring(() => ({
  // Your spring props go here...
  onRest: () => {
    lettersApi.resume();
  },
}));
```

---

Alternatively, if you want some overlapping of the animations, just like you can do with `useChain`'s `timeSteps` array, you can use the same approach as the example above, but now chain the animations using the `onStart` event callback with a vanilla JavaScript `setTimeout` - just experiment with the milliseconds.
```js
const [trail, trailApi] = useTrail(trailLength, () => ({
  // Your trail props go here...
  pause: true,
}));

const [springStyles, springApi] = useSpring(() => ({
  // Your spring props go here...
  onStart: () => {
    // Resume (play) the trail animation 200ms after starting the spring animation
    setTimeout(() => {
      trailApi.resume();
    }, 200);
  },
}));
```

**Comment by joshuaellis** [comment]
This is sort of to do with our `useTrail` implementation where the trail implementation doesn't work with `SpringRef` which can be seen when you just call `ref.start`... Although in `useChain` we start the `Controllers` directly, except we're not re-attaching springs to other controllers, this logic is all locked up in `useTrail`. 

To solve it'll require more thought and re-writing, maybe we introduce a flag or prop similar to `useTransition`.

**Comment by joshuaellis** [comment]
I've released this on `v9.4.0-beta.5` if anyone wants to try it out and feedback that would be really useful, I did cross check it with previous behaviour in v8, so I'm hoping this is as expected.

**Comment by phmasek** [comment]
@joshuaellis Given your fix, is the implementation provided by @mrgnou expected to work or is there another solution for making this work?

Seems like using `useTrail` together with `useSpring` through `useChain` with timeSteps does not seem to delay the `useTrail` animation.
